src/embeddings/tools/embedding_vocab.cpp

```diff
@@ -51,7 +84,15 @@ int main(int argc, char** argv)
             stream->set_content(analyzers::get_content(doc));
 
             while (*stream)
+            {
                 ++vocab[stream->next()];
+
+                if (algorithm == SGNS && (vocab.size() >= min_reduce_threshold * max_size))
```

Comment:
Thanks for taking the time to understand what was going on and provide in-depth explanation. 

I've taken this to mean all the way back before we made any changes and reverted this file.